tests/network/dry_run/test_dry_run_kubemacpool.py

```diff
@@ -130,11 +117,12 @@ def dry_run_vm_with_mac_address(
 def test_dry_run_mac_not_saved(
     dry_run_vma_mac_address,
     vm_with_mac_address,
-    running_vm_mac_addresses,
 ):
     """Assure that the MAC address given to a VM created in dry-run mode is still available in the MAC-pool."""
-    assert dry_run_vma_mac_address == running_vm_mac_addresses, (
-        f"{vm_with_mac_address.name} MAC address is {running_vm_mac_addresses}"
+    vm_interfaces = vm_with_mac_address.get_interfaces()
```

Comment:
I don't see that we start the VM, was this step removed on purpose?

Reply:
I expressed the change in the commit message:
> Running a VM in the KMP dry-run test is redundant as the test is just
checking that the MAC address is assigned by KMP. Running the VM is not
really required as it is not checking any KMP functionality.

Reply:
@Anatw ?

Reply:
If you don't start it, how can you know that the address is available? If the VM isn't started, we can't be sure the assigned MAC address is actually usable. While KMP allocates a MAC when the VMI is defined, I don’t believe it performs full conflict checks or enforcement until the VM is launched. Starting the VM ensures that the MAC address survives the full validation path.

Reply:
This is not how KMP works and it has nothing to do with a running VM.
KMP works on the VM and not on the VMI, therefore there is no meaning to start a VM.
Therefore, the fact a VM was deployed is enough to check KMP added the mac address, and this is exactly what is done here.